internal/utils/index/konnectcloudgatewaydataplanegroupconfiguration.go

```diff
@@ -21,5 +23,28 @@ func OptionsForKonnectCloudGatewayDataPlaneGroupConfiguration(cl client.Client)
 			Field:          IndexFieldKonnectCloudGatewayDataPlaneGroupConfigurationOnKonnectGatewayControlPlane,
 			ExtractValueFn: indexKonnectGatewayControlPlaneRef[konnectv1alpha1.KonnectCloudGatewayDataPlaneGroupConfiguration](cl),
 		},
+		{
+			Object:         &konnectv1alpha1.KonnectCloudGatewayDataPlaneGroupConfiguration{},
+			Field:          IndexFieldKonnectCloudGatewayDataPlaneGroupConfigurationOnKonnectNetworkRef,
+			ExtractValueFn: konnectCloudGatewayNetworkDataPlaneGroupConfigurationRef,
+		},
+	}
+}
+
+func konnectCloudGatewayNetworkDataPlaneGroupConfigurationRef(object client.Object) []string {
+	dpg, ok := object.(*konnectv1alpha1.KonnectCloudGatewayDataPlaneGroupConfiguration)
+	if !ok {
+		return nil
 	}
+
+	return lo.Map(
```

Comment:
I think using `lo.FilterMap` would make it more readable and more effective.

Reply:
:+1 21311c28bf8a67b8113988e5eca09d7a560119e2